### PR TITLE
feat: Emit events to the devserver process when the spec list is changed

### PR DESCRIPTION
**Pull request by JessicaSachs**
This PR creates the initial Dev Server plugin and framework for sending emitting events across processes. The first event is to notify the webpack plugin and frontend that new specs were added or removed.

### CODE REVIEW

File: packages/server-ct/src/routes-ct.js [hunk 1]
```diff
@@ -12,7 +12,7 @@ module.exports = ({ app, config, project, onError }) => {
   app.get('/__cypress/runner/*', runnerCt.middleware(send))
 
   app.get('/__cypress/static/*', (req, res) => {
-    staticPkg.handle(req, res)
+    staticPkg.middleware(req, res)
```
Comment (lmiller1990) [inline packages/server-ct/src/routes-ct.js:15]: Nice, noticed that `handle` doesn't exist on this module

File: packages/server/lib/plugins/devserver.js [hunk 2]
```diff
@@ -1,141 +1,37 @@
...
 
 // for simpler stubbing from unit tests
 const API = {
-  // errorMessage,
-
-  // clientSideError,
-
   emitter: baseEmitter,
 
-  getFile (filePath, config) {
```
Comment (lmiller1990) [inline packages/server/lib/plugins/devserver.js:59]: deleted a lot of code here - this definitely isnt used anywhere else?

File: packages/server-ct/src/routes-ct.js (hunk 1 above)
Comment (JessicaSachs) [reply packages/server-ct/src/routes-ct.js:15]: Yeah, I don't know where it went 😬... I should ping someone from TR about it @flotwig or @Bkucera perhaps?

File: packages/server/lib/plugins/devserver.js (hunk 2 above)
Comment (JessicaSachs) [reply packages/server/lib/plugins/devserver.js:59]: No, this was an initial copy paste from preprocessor.js. I created devserver.js in a previous commit.

File: packages/server-ct/src/routes-ct.js (hunk 1 above)
Comment (flotwig) [reply packages/server-ct/src/routes-ct.js:15]: Not familiar with this, but lol

Comment (flotwig) [reply packages/server-ct/src/routes-ct.js:15]: Looks like it exists currently? https://github.com/cypress-io/cypress/blob/a4a3c9eb7d1d6941ba71dd4deb1eb640e8a37a62/packages/server/lib/controllers/static.js#L1-L12

Comment (flotwig) [reply packages/server-ct/src/routes-ct.js:15]: https://github.com/cypress-io/cypress/blob/a4a3c9eb7d1d6941ba71dd4deb1eb640e8a37a62/packages/server/lib/routes.js#L38-L42

Comment (elevatebart) [reply packages/server-ct/src/routes-ct.js:15]: It came from a bad merge. 
The `handle` function was coming from `packages/server/lib/controlloers/static`.
Which was in turn using package/static.
Now that we are using `package/static` directly, it does not make sense anymore.

Thank you @lmiller1990 for fixing it. ;)

**Review by lmiller1990 — APPROVED** [review]
✅

**Review by lmiller1990 — APPROVED** [review]

File: packages/server-ct/src/routes-ct.js (hunk 1 above)
Comment (JessicaSachs) [reply packages/server-ct/src/routes-ct.js:15]: Yikes. I'll fix it. Thanks, that merge was super hairy.

**Pull request merged**